### PR TITLE
feat(build): scan translations for invalid attributes

### LINKED ISSUE #61: Move Grunt XSS tasks to the l10n repo
Per @vladikoff's https://github.com/mozilla/fxa-auth-mailer/issues/67#issuecomment-125216940

> _if the "full" templates live in the l10n repo then we should probably move those XSS parse tasks out of the content server and put them into the l10n repo_

### PR DESCRIPTION
**Pull request by shane-tomlinson**
* The allowed list of attributes is generated from the .pot files in locale/templates/*
* A list of expected attributes and values are saved.
* When xss-parse is run, all attributes in the translated files are checked
  against the allowed list.
* no intermediate representations are used, extracting allowed attributes and examining translations are all done against the original .po and .pot files.

fixes #61

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@zaach, @vladikoff, and @rfk - r?

**Comment by shane-tomlinson** [comment]
fixes #61

**Comment by vladikoff** [comment]
@shane-tomlinson does it make sense to add Travis CI with this? so it scans when there is a new build?

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson does it make sense to add Travis CI with this? so it scans when there is a new build?

It does.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+language: node_js
+
+node_js:
+  - "0.12"
```
Comment (vladikoff) [inline .travis.yml]: awww yeaaaaaa

**Comment by shane-tomlinson** [comment]
Not ready yet! I just found some issues.

File: .travis.yml (hunk 1 above)
Comment (pdehaan) [inline .travis.yml]: what, no Node v4 support?

Comment (vladikoff) [inline .travis.yml]: We can use
```
node_js:
  -  "4"
```

**Comment by rfk** [comment]
@shane-tomlinson did we decide on whether it was safe for the logic to live in this repo, where push access may be less tightly controlled than other project repos?

**Comment by rfk** [comment]
I really like how simple this seems.  Do you have thoughts on hooking it up to some automated testing so we can be sure it continues to work after we stop paying attention to it?

**Comment by shane-tomlinson** [comment]
> I really like how simple this seems. Do you have thoughts on hooking it up to some automated testing so we can be sure it continues to work after we stop paying attention to it?

As part of this PR, it's hooked up to travis. :D

**Comment by shane-tomlinson** [comment]
> As part of this PR, it's hooked up to travis. 

We should probably have travis failures notify us directly.

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson did we decide on whether it was safe for the logic to live in this repo, where push access may be less tightly controlled than other project repos?

Yeah, I want to remove the from this repo and into its own repo, and sooner rather than later.

**Comment by rfk** [comment]
> thoughts on hooking it up to some automated testing

I also mean, how do we test that this code itself is working, and will actually detect problems if they show up in the translations?

**Comment by shane-tomlinson** [comment]
> I also mean, how do we test that this code itself is working, and will actually detect problems if they show up in the translations?

I'm working on a branch for just that problem.

**Comment by shane-tomlinson** [comment]
Ref https://github.com/mozilla/grunt-l10n-lint, where the lint logic now lives.

**Comment by rfk** [comment]
Is this ready to have "WIP" removed and find a formal reviewer?

**Comment by shane-tomlinson** [comment]
> Is this ready to have "WIP" removed and find a formal reviewer?

Yes.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+cache:
+  directories:
+    - node_modules
+
+notifications:
+  irc:
+    channels:
+      - 'irc.mozilla.org#fxa-bots'
+    use_notice: false
+    skip_join: false
```
Comment (rfk) [inline .travis.yml]: This seems important enough what we should add email notifications on failure, they'd be easily missed in IRC.

File: package.json [hunk 3]
```diff
@@ -13,5 +13,11 @@
   "license": "MPL-2.0",
   "bugs": {
     "url": "https://github.com/mozilla/fxa-content-server-l10n/issues"
+  },
+  "devDependencies": {
+    "grunt": "0.4.5",
+    "grunt-l10n-lint": "0.0.0",
```
Comment (rfk) [inline package.json]: I'm surprised and impressed that npm lets you register this as a package version....

**Comment by rfk** [comment]
r+ on this PR; I've also added the grunt-l10n-lint repo to our waffleboard for any follow-up bugs/comments.

File: .travis.yml (hunk 2 above)
Comment (shane-tomlinson) [inline .travis.yml]: > This seems important enough what we should add email notifications on failure, they'd be easily missed in IRC.

Added.

**Pull request merged**